docs/network-policy.md

```diff
@@ -0,0 +1,365 @@
+# NetworkPolicy
+
+Kubernetes NetworkPolicy documentation: https://kubernetes.io/docs/concepts/services-networking/network-policies
+
+Kubernetes NetworkPolicy API reference: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.19/#networkpolicy-v1-networking-k8s-io
+
+By default K8's pods may send to any recipient and receive from any source. When a pod is selected by one or more NetworkPolicies it becomes isolated and all traffic is blocked. NetworkPolicy may then be used to allow received traffic from sources determined by the `ingress` section and/or send traffic to recipients specified by the `egress` section.
+
+Each NetworkPolicy object consists of four sections:
+
+1. `podSelector`: a label selector that determines which pods the NetworkPolicy applies to
+2. `policyTypes`: determines which policy tpes are included, if none are selected then `Ingress` will always be set and `egress` will be set if any `egress` rules are applied 
+3. `ingress rules`: determines the sources that pods selected by the ingress rule can receive traffic from 
+4. `egress rules`: determines the sinks that pods selected by the egress rule can send trafic to 
+
+# NetworkPolicy features 
+
+These are descibed in order and are additive 
+
+## **Unicast default-deny**
+
+When a pod is selected by one or more NetworkPolicies, the `policyTypes` is set to both `ingress` and `egress`, and if no rules are specified it becomes isolated and all unicast ingress and egress traffic is blocked. 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+If only ingress traffic to all pods in a namespace needs to be blocked the following can be used 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+```
+
+And finally if only egress traffic to all pods in a namespace needs to be blocked 
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
```

Comment:
Is this supposed to be Egress or is line 46 supposed to be?

Reply:
ahh thanks, it's supposed to be Egress